csrc/mmdeploy/codebase/mmseg/segment.cpp

```diff
@@ -44,6 +51,29 @@ class ResizeMask : public MMSegmentation {
     OUTCOME_TRY(auto host_tensor, MakeAvailableOnDevice(mask, host, stream_));
     OUTCOME_TRY(stream_.Wait());
 
+    if (!with_argmax_ && mask.shape(1) > 1 && mask.shape(1) == classes_ &&
+        host_tensor.data_type() == DataType::kFLOAT) {
+      int stride = height * width;
+      Tensor mask_out = TensorDesc{Device("cpu"),
+                                   DataType::kFLOAT,
+                                   {mask.shape(0), 1, mask.shape(2), mask.shape(3)},
+                                   "argmax_out"};
+      auto ptr = host_tensor.data<float>();
+      auto out_ptr = mask_out.data<float>();
+      for (int i = 0; i < stride; i++, ptr++) {
+        auto v = *ptr;
+        auto idx = 0.f;
```

Comment:
Isn't `idx` an integer?

Reply:
Yes， I tried to make it an int value but found that I have to make do `host_tensor = mask_out`. Or the program would crash when drawing masks in the demo.